databases/week3/databases_week3.sql

```diff
@@ -0,0 +1,209 @@
+
+CREATE TABLE reservation (
+id INT UNSIGNED NOT NULL AUTO_INCREMENT,
+number_of_guests INT UNSIGNED NOT NULL,
+meal_id INT UNSIGNED NOT NULL,
+created_date DATETIME NOT NULL DEFAULT NOW(),
+PRIMARY KEY (id),
+FOREIGN KEY (meal_id) REFERENCES meal (id) ON DELETE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE review (
+id INT UNSIGNED NOT NULL AUTO_INCREMENT,
+title VARCHAR(255) NOT NULL,
+description TEXT NOT NULL,
+meal_id INT UNSIGNED NOT NULL,
+stars INT UNSIGNED NOT NULL,
+created_date DATETIME NOT NULL DEFAULT NOW(),
+PRIMARY KEY (id),
+FOREIGN KEY (meal_id) REFERENCES meal (id) ON DELETE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- CREATING queries
+-- MEAL
+-- Get all meals
+SELECT *
+FROM meal;
+-- Add a new meal
+
+INSERT INTO meal (
+title,
+description,
+location,
+`when`, 
+max_reservations,
+price
+)
+VALUES
+(
+'Hamburger', 
+'sandwich consisting of one or more cooked patties of ground meat, usually beef, placed inside a sliced bread roll or bun', 
+'Rådhuspladsen',
+'2021-04-26',
+4,
+80
+);
+-- Get a meal with any id, fx 1
+SELECT * 
+FROM meal
+WHERE id = 1;
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE meal SET title = 'Ham cheese burger'
+WHERE meal.id = 1;
+
+ -- Delete a meal with any id, fx 1
+ 
+DELETE FROM meal
+WHERE id = 1;
+
+-- RESERVATION:
+
+-- Get all reservations
+
+SELECT * 
+FROM reservation;
+
+-- Add a new reservation
+
+INSERT INTO reservation (
+number_of_guests,
+meal_id,
+created_date
+)
+VALUES
+(
+10, 
+2,
+'2021-04-27'
+);
+-- Get a reservation with any id
+
+SELECT * 
+FROM reservation
+WHERE id = 2;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE reservation 
+SET number_of_guests = 20
+WHERE reservation.id = 2;
+
+-- Delete a reservation with any id, fx 1
+
+DELETE 
+FROM reservation
+WHERE id = 2;
+
+-- Review
+-- Get all reviews
+SELECT * 
+FROM review;
+
+-- Add a new review
+
+INSERT INTO review (
+title,
+description,
+meal_id,
+stars,
+created_date
+)
+VALUES
+(
+'excellent',
+'great food, excellent service and nice staff', 
+2,
+4,
+'2021-04-27'
+);
+
+-- Get a review with any id, fx 1
+
+SELECT * 
+FROM review
+WHERE id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE review SET stars = 4.5
+WHERE id = 1;
+
+-- Delete a review with any id, fx 1
+
+DELETE 
+FROM review
+WHERE id = 1;
+
+-- Additional queries
+
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT title AS meal_title, description AS meal_description, 
+price AS meal_price 
+FROM meal
+WHERE price < 90;
+
+-- Get meals that still has available reservations
+
+SELECT meal.id AS meal_id, meal.title AS meal_title,  
+SUM(reservation.number_of_guests), 
+meal.max_reservations
+FROM meal 
+JOIN reservation ON reservation.meal_id = meal.id
```

Comment:
What if a meal doesn't have reservations yet?

Also, once you address that, you might want to take a look at this function: COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce (if you have meals without reservations you'll have nulls).